### PR TITLE
Reconcile Argo applications

### PR DESCRIPTION
**Pull request by kubasobon**
:construction: :construction_worker: Subject to rapid change. Spiking Argo Application support in `release-operator`. :construction: 

## Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/key/key.go [hunk 1]
```diff
@@ -160,6 +160,7 @@ func GetProviderOperators() []string {
 	return []string{ProviderOperatorAWS, ProviderOperatorAzure, ProviderOperatorKVM}
 }
 
+//TODO(kuba): check if used; remove if not
```
Comment (kubasobon) [inline service/controller/key/key.go:163]: `key` cleanup is not something I wanted to start with.

File: service/controller/release/resource/argoapps/configversion/service.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	}
+
+	return s, nil
+}
+
+// Get tries to resolve config version for the app version.
+//
+// It returns error matched by IsNotFound if the app version does not have
+// "config.giantswarm.io/version" annotation in the catalog.
+func (s *Service) Get(ctx context.Context, ac argoapp.ApplicationConfig) (string, error) {
```
Comment (kubasobon) [inline service/controller/release/resource/argoapps/configversion/service.go:43]: This entire service comes from `config-controller`.

**Comment by kubasobon** [comment]
@giantswarm/team-biscuit Please take a peek

**Comment by kopiczko** [comment]
LMK when I should have a look

**Comment by kubasobon** [comment]
I'm abandoning this PR in favour of a new, better one.

**Pull request closed without merging**